### PR TITLE
Shopify: Fill in recommended Product schema jsonld fields for SEO

### PR DESCRIPTION
**Pull request by masonmcelvain**
Injects the new jsonld we are building on the backend (see https://github.com/iFixit/ifixit/pull/40347) into the html `<head>`. We don't need to build the jsonld in JS anymore, so we can drop that code.

Also updates some testing data and a snapshot test.

## QA
Start the dev server and make sure the jsonld script tag gets added IFF the jsonld is not empty. Copy the `<head>` into [google's rich results tool](https://search.google.com/test/rich-results) and make sure there are no warnings or errors.

CC: @sterlinghirsh @JFleischhauer @GrantMatejka

Closes https://github.com/ifixit/ifixit/issues/39560

### CODE REVIEW

**Comment by cdcline** [comment]
CR 👍

**Comment by masonmcelvain** [comment]
The parent pull merged so in theory we should be able to test this now

File: src/lib/product_reviews/visualizer/index.js [hunk 1]
```diff
@@ -89,7 +109,16 @@ class Visualizer extends Component {
            "ratingValue": average,
            "reviewCount": numReviews
          },
-         "review": [schemaReviews],
+         "brand": {
+            "@type": "Brand",
+            "name": "iFixit",
```
Comment (GrantMatejka) [inline src/lib/product_reviews/visualizer/index.js:114]: Will this be true for every product that uses this component?

**Comment by masonmcelvain** [comment]
The parent pull failed on deploy so we can't QA this yet

File: src/lib/product_reviews/visualizer/index.js (hunk 1 above)
Comment (GrantMatejka) [reply src/lib/product_reviews/visualizer/index.js:114]: Yeah me neither, but if we don't have the data we should probably just not include it

**Comment by zdmitchell** [comment]
CR 🎸 but deploy_block 🎸 on checking with someone about https://github.com/iFixit/product-page-components/pull/41#discussion_r733124854.

**Comment by masonmcelvain** [comment]
QA ✌🏻 The jsonld gets injected into the < head> when you run the dev server, and copying the html into [Google's Rich Results Test tool](https://search.google.com/test/rich-results) passes with no warnings or errors.

- Ran `npm start`
- Inspected the < head> element and saw that only the nonempty jsonld was added to the head
image
- Copied the contents of the < head> into the rich results test, and confirmed the test passed
image

Now that we've confirmed it looks like we want, I'm going to update the snapshot tests.

**Comment by masonmcelvain** [comment]
QA ✌🏻 carry forward, just updated a test

**Comment by masonmcelvain** [comment]
un_dev_block ✌🏻 

un_deploy_block ✌🏻 We worked out where 'brand' data gets sourced from in iFixit/ifixit#40347

**Comment by cdcline** [comment]
CR 👍

File: src/lib/product_reviews/visualizer/index.js [hunk 2]
```diff
@@ -60,50 +60,16 @@ const ReviewCount = styled.h3`
...
-            <Helmet>
-               {numReviews > 0 &&
-                  <script type="application/ld+json">{JSON.stringify(reviewJsonld)}</script>
-               }
-            </Helmet>
+            {Object.keys(jsonld).length && (
+               <Helmet>
+                  <script type="application/ld+json">{JSON.stringify(jsonld)}</script>
+               </Helmet>
+            )}
```
Comment (zdmitchell) [inline src/lib/product_reviews/visualizer/index.js:72]: Doesn't React output 0 here if length is 0?

**Comment by zdmitchell** [comment]
CR 🎸 but dev_block 🎸 on what I think might be a bug.

File: src/lib/product_reviews/visualizer/index.js (hunk 2 above)
Comment (masonmcelvain) [reply src/lib/product_reviews/visualizer/index.js:72]: You're totally right, thanks for catching that

Comment (masonmcelvain) [reply src/lib/product_reviews/visualizer/index.js]: https://github.com/iFixit/product-page-components/pull/41/commits/b7454f3f0f26b25b06aa0f7bb14828dd75de189f

**Comment by zdmitchell** [comment]
CR 🎸

**Comment by cdcline** [comment]
CR 👍

**Pull request merged**